spec/rspec/matchers/built_in/respond_to_spec.rb

```diff
@@ -493,6 +515,16 @@ def obj.some_method(arg, arg2, arg3, *args); end
       expect(obj).to respond_to(:foo).with_keywords(:a, :b)
     end
 
+    it 'passes if target is :new with keywords' do
+      # note we can't use the metaobject definition for initialize
+      klass = eval %{Class.new { def initialize(a: nil, b: nil); end}}
+      expect(klass).to respond_to(:new).with_keywords(:a, :b)
+
+      # note we can't use the metaobject definition for initialize
+      klass_2 = eval %{Class.new { def initialize(**kw_args); end}}
+      expect(klass_2).to respond_to(:new).with_keywords(:a, :b)
+    end
+
```

Comment:
We need some specs for when `new` has been monkeyed with.  See the 3 contexts starting here:

https://github.com/rspec/rspec-mocks/pull/886/files#diff-7a6cccf0d2c37e75e4affc0a24501df6R426

It looks like your implementation handles this but you haven't written specs for it.